bin/resty

```diff
@@ -47,6 +66,8 @@ GetOptions("c=i",             \(my $conns_num),
            "V|v",             \(my $version))
    or usage(1);
 
+check_arg_format('--shdict', qr/^[a-z]*\s\d+(?i)[km]$/, 'NAME SIZE', @shdicts);
```

Comment:
This regex looks wrong to me. Some notes:

1. The shared dict name is mandatory while `*` allows empty names.
2. The shared dict name can contain underscores.
3. The size units `k` and `m` are optional. default to bytes.
4. we should allow more than more spaces between `NAME` and `SIZE`.
5. `\z` is better than `$` here to avoid multiple lines in the value.

Also, it's better to pass the `@shdicts` array as an array reference, that is, `\@shdicts`.